src/compiler/core.ts

```diff
@@ -787,50 +869,83 @@ namespace ts {
         return keys;
     }
 
-    /**
-     * Enumerates the properties of a Map<T>, invoking a callback and returning the first truthy result.
-     *
-     * @param map A map for which properties should be enumerated.
-     * @param callback A callback to invoke for each property.
-     */
-    export function forEachProperty<T, U>(map: Map<T>, callback: (value: T, key: string) => U): U {
-        let result: U;
-        for (const key in map) {
-            if (result = callback(map[key], key)) break;
+    function arrayFrom<T>(iterator: Iterator<T>): T[] {

```

Comment:
Maybe we should just export this and replace all references to `keysOfMap` and `valuesOfMap`.